Add dictionaries directory builder

mrtcode opened the pull request.

mrtcode committed:
Add dictionaries directory builder

dstillman commented on dictionaries/build-dictionaries:
```diff
@@ -0,0 +1,31 @@
+import urllib.request, json
+
+dictionaries = []
+with urllib.request.urlopen("https://services.addons.mozilla.org/api/v4/addons/language-tools/?app=firefox&type=dictionary") as url:
```
`url` → `resp` or something, here and for the other `urlopen` call. (I would always expect `url` to be a URL string, as it is in the other instance.)

dstillman committed:
Dictionary builder improvements

- Filter list to dictionaries best matching specific locales (using
  client locale list plus some extras), and download only those files
- Include native dictionary name in JSON for use by client
- Save a list of unused dictionaries for review

dstillman commented:
The script will now generate a filtered list of just the dictionaries the client should offer, which includes any dictionaries found for the client locales plus a few extras that seemed worthwhile:

https://zotero-download.s3.amazonaws.com/dictionaries/dictionaries.json
https://zotero-download.s3.amazonaws.com/dictionaries/dictionaries-unused.json
